src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Cohort.cs

```diff
@@ -44,18 +46,38 @@ public Cohort()
         public string AccountLegalEntityPublicHashedId { get; set; }
         public Originator Originator { get; set; }
 
-        public virtual ICollection<Apprenticeship> Apprenticeship { get; set; }
-        public virtual ICollection<Message> Message { get; set; }
-        public virtual ICollection<TransferRequest> TransferRequest { get; set; }
+        public virtual ICollection<Apprenticeship> Apprenticeships { get; set; }
+        public virtual ICollection<Message> Messages { get; set; }
+        public virtual ICollection<TransferRequest> TransferRequests { get; set; }
 
-        public virtual void AddDraftApprenticeship(DraftApprenticeshipDetails draftApprenticeshipDetails)
+        [NotMapped]
+        public IEnumerable<DraftApprenticeship> DraftApprenticeships => Apprenticeships.OfType<DraftApprenticeship>();
+
+        public virtual void AddDraftApprenticeship(DraftApprenticeshipDetails draftApprenticeshipDetails, Originator modifyingParty)
         {
+            EnsureModifierIsAllowedToModifyDraftApprenticeship(modifyingParty);
             ValidateDraftApprenticeshipDetails(draftApprenticeshipDetails);
             var draftApprenticeship = new DraftApprenticeship(draftApprenticeshipDetails, Originator);
-            Apprenticeship.Add(draftApprenticeship);
+            Apprenticeships.Add(draftApprenticeship);
             Publish(() => new DraftApprenticeshipCreatedEvent(draftApprenticeship.Id, Id, draftApprenticeship.Uln, draftApprenticeship.ReservationId.Value, CreatedOn.Value));
         }
 
+        public virtual void UpdateDraftApprenticeship(DraftApprenticeshipDetails draftApprenticeshipDetails, Originator modifyingParty)
+        {
+            EnsureModifierIsAllowedToModifyDraftApprenticeship(modifyingParty);
+
+            ValidateDraftApprenticeshipDetails(draftApprenticeshipDetails);
+            var existingDraftApprenticeship = DraftApprenticeships.SingleOrDefault(a => a.Id == draftApprenticeshipDetails.Id);
+
+            if (existingDraftApprenticeship == null)
+            {
+                throw new InvalidOperationException($"There is not a draft apprenticeship with id {draftApprenticeshipDetails.Id} in cohort {Id}");
+            }
+            
+            existingDraftApprenticeship.Merge(draftApprenticeshipDetails);
```

Comment:
Just checking for my understanding - is it ok to allow `ReservationId` to be changed here?

Reply:
We don't know at this point in the code do we? If the web-app supplies a new reservation id why _wouldn't_ we persist it?

Reply:
When updating a draft apprenticeship can a new reservation ID be chosen?